lab/tree_builder.py

```diff
@@ -13,21 +14,35 @@ def __init__(self, name, val):
 class TreeBuilder(object):
 
     def __init__(self, root):
-        # root will be type TreeNode
         self.root = root
         self.size = 0
 
     # Function to push to tree 
     def push(self, node):
-        # TODO
-        pass
+        currentNode = self.root
+        while True:
+            if node.val < currentNode.val:
+                if currentNode.left:
+                    currentNode = currentNode.left
+                else:
+                    currentNode.left = node
+                    node.parent = currentNode
+                    self.size += 1
+                    break
+            else:
+                if currentNode.right:
+                    currentNode = currentNode.right
+                else: 
+                    currentNode.right = node
+                    node.parent = currentNode
+                    self.size += 1
+                    break
 
     # Function to remove node from tree 
     def pop(self, name):
-        # TODO
-        pass
+        node = find_by_attr(self.root, name)
+        node.parent = None
```

Comment:
This fails when the node does not exist in the tree. For example in this case:
```python
lb = TreeNode("Lebron James", 12.3)
kb = TreeNode("Kobe Bryant", 24.3)
mj = TreeNode("Michael Jordan", 14.5)
sp = TreeNode("Scottie Pippen", 11.6)
tree = TreeBuilder(lb)
tree.push(kb)
tree.push(mj)
tree.push(sp)
tree.pop("No Name")
```

Met with:
```bash
$ python visualize.py 
Traceback (most recent call last):
  File "visualize.py", line 31, in <module>
    main()
  File "visualize.py", line 28, in main
    tree.pop("No Name")
  File "~/projects/NBA-Search/lab/tree_builder.py", line 44, in pop
    node.parent = None
AttributeError: 'NoneType' object has no attribute 'parent'
```

Let's handle this issue and also return a boolean whether or not the node was successfully removed. In the above example the function should accurately return `False`.